doc/adrs/198_add_the_ability_to_share_a_url_containing_the_views_of_the_workbench.adoc

```diff
@@ -2,325 +2,164 @@
-        {
-          "id": "details",
-          "configuration": { ... }
-        },
-        {
-          "id": "query",
-          "configuration": { ... }
-        },
-        {
-          "id": "representations",
-          "configuration": { ... }
-        },
-        {
-          "id": "related_elements",
-          "configuration": { ... }
-        }
-      ],
-      "configuration": {
-        "partId": "details",
-        "closed": false
-      }
-    }
-  ],
-  "focus": "center"
+interface WorkbenchPartShareableState {
+  parts?: { [partId: string]: WorkbenchPartShareableState }
 }
 ----
 
-Notes:
-* Since each `WorkbenchPart` has its own configuration, the default configurations for the various views are omitted for now. In the implementation, we will have to make explicit, for each view that wants to be able to contribute to the URL (`Explorer`, `Validation`, `Details`, `Query`, `Representations`, `Related Elements` and all representation editors) the default configuration to use.
-* In the Sirius Web application, the "left" and "right" panels are essentially similar, so their configurations follow the same structure:
+where:
 
-[source, json]
-----
-interface SiriusWebApplicationSidePanelConfiguration {
-	"partId": string,
-	"closed": boolean
-}
-----
+* `WorkbenchPartShareableState` is a recursive data structure used to capture the shareable state of the workbench. This structure mimics the structure of the main React components used to implement the workbench frontend.
+* `WorkbenchPartShareableState.parts` is a map capturing the shareable state of the children workbench parts.
+* `WorkbenchPartShareableState.parts` is optional so that a part without children can be represented by not having a value for parts, thus saving URL real estate.
+** The key `partId` is a string that can be used to uniquely identify, at runtime, a part in the parent part.
+It is _not_ the runtime id of the React Component corresponding to the part.
+It is a predictable and unambiguous id given by the parent part (or by the application for the root part) at creation time.
+** The value is a `WorkbenchPartShareableState` representing the shareable state of the child part.
 
-where:
-** `SiriusWebApplicationSidePanelConfiguration.partId: string` encodes the ID of the `WorkbenchPart` in this panel that is active ("left" default: "explorer" ; "right" default: "details").
-** `SiriusWebApplicationSidePanelConfiguration.closed: boolean` encodes whether the `WorkbenchPart` is opened or closed (default for both: "false").
+Note: A follow-up ADR will focus on the possibility to share the internal state of a part (e.g. which filters of the 'Explorer' view are enabled), but this is out of the scope of this ADR.
 
+The main impacts on the Sirius Web frontend components are:
+* `partId` must be added to the props of the components.
+* A ref is passed to the components so that they are able to provide their shareable state when the shareable link is to be created.
```

Comment:
That's not need here, the internal state of a part is out of scope, you said it yourself